### PR TITLE
HPCC-19014 Implement common method to query shared HPCC key filespecs

### PR DESCRIPTION
**Pull request by RussWhitehead**
Create new environment.conf keywords to specify general purpose PKI keyfiles.
Create a common, shared JLIB method to retrieve these settings, and deprecate
the legacy DFU specific one.

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [x] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@mckellyln  Please review

File: system/jlib/jutil.cpp [hunk 1]
```diff
@@ -2505,6 +2528,23 @@ jlib_decl bool queryDafsSecSettings(DAFSConnectCfg *_connectMethod,
...
 }
 
+//query PKI values from environment.conf
+jlib_decl bool queryHPCCPKIKeyFiles(const char * *  _certificate,//HPCCCertFile
+                                    const char * *  _privateKey, //HPCCPrivateKeyFile
+                                    const char * *  _passPhrase) //HPCCPassPhrase
+{
+    bool rc = true;
+    if (HPCCcertificate.isEmpty() && HPCCprivateKey.isEmpty() && HPCCpassPhrase.isEmpty())
+        rc = querySecuritySettings(nullptr, nullptr, nullptr, nullptr, nullptr);
```
Comment (mckellyln) [inline system/jlib/jutil.cpp]: querySecuritySettings() checks for dfsUseSSL which does not seem to matter for a general purpose routine.
Can we have a more general routine that both the old querySecuritySettings() and the new routine could call to get the Certificate/PrivateKeyFile/PassPhrase values ?

**Comment by mckellyln** [comment]
@RussWhitehead one comment about adding a new, more general routine to get the CertFile/PrivateKey/PassPhrase and having the old querySecuritySettings perhaps use that.

**Comment by RussWhitehead** [comment]
@mckellyln Please rereview

**Comment by mckellyln** [comment]
Move this
{noformat}
               sslMethod.set(conf->queryProp("dfsUseSSL"));
{noformat}
and its associated logic into the querySecuritySettings(), so the
new method has only what it needs and the querySecuritySettings() has this dfsUseSSL logic and then calls the new method to get the file info.

**Comment by RussWhitehead** [comment]
The connectMethod is needed in both querySecuritySettings() and queryDafsSecSettings(), so IMHO it should only be deduced once, in retrieveSecuritySettings()

**Comment by RussWhitehead** [comment]
@mckellyln Please rereview

File: system/jlib/jutil.cpp [hunk 2]
```diff
@@ -2497,14 +2484,40 @@ jlib_decl bool queryDafsSecSettings(DAFSConnectCfg *_connectMethod,
...
+    const IProperties & conf = queryEnvironmentConf();
+    if (_certificate)
+        *_certificate = conf.queryProp("HPCCCertFile");
+    if (_privateKey)
+        *_privateKey = conf.queryProp("HPCCPrivateKeyFile");;
+    if (_passPhrase)
+    {
+        if (HPCCpassPhrase.isEmpty())
+        {
+		    const char *passPhrasePtr = conf.queryProp("HPCCPassPhrase");
```
Comment (mckellyln) [inline system/jlib/jutil.cpp]: Add a CriticalSection to prevent collision

File: system/jlib/jutil.cpp [hunk 3]
```diff
@@ -2401,92 +2401,79 @@ jlib_decl const IProperties &queryEnvironmentConf()
...
-    else
-    {
-        if (_connectMethod)
-            *_connectMethod = SSLNone;
-        if (_port)
-            *_port = DAFILESRV_PORT;
+		{
+			if (DAFSpassPhrase.isEmpty())
+			{
+				const char *passPhrasePtr = conf.queryProp("dfsSSLPassPhrase");
```
Comment (mckellyln) [inline system/jlib/jutil.cpp]: CriticalSection here as well ?

File: system/jlib/jutil.cpp [hunk 4]
```diff
@@ -2401,92 +2401,79 @@ jlib_decl const IProperties &queryEnvironmentConf()
...
+            else // SSLNone or false or ...
+                *_connectMethod = SSLNone;
         }
-    }
-    if (retrieved)
-    {
-        if (_connectMethod)
-            *_connectMethod = connectMethod;
-        if (_port)
+
```
Comment (mckellyln) [inline system/jlib/jutil.cpp]: Start deprecated dfs certificate code comment ...

File: system/jlib/jutil.cpp [hunk 5]
```diff
@@ -2401,92 +2401,79 @@ jlib_decl const IProperties &queryEnvironmentConf()
...
+				if (!isEmptyString(passPhrasePtr))
+		        {
+		            dfsKeywords = true;
+					decrypt(DAFSpassPhrase, passPhrasePtr);
+		        }
+			}
+			*_passPhrase = DAFSpassPhrase.str();
+		}
+
+		if (!dfsKeywords && (_certificate || _privateKey || _passPhrase))
```
Comment (mckellyln) [inline system/jlib/jutil.cpp]: End deprecated dfs certificate comment

File: system/jlib/jutil.cpp [hunk 6]
```diff
@@ -2401,92 +2401,79 @@ jlib_decl const IProperties &queryEnvironmentConf()
...
+		{
+			// port to try first (or only) ...
+			if (!_connectMethod || *_connectMethod == SSLNone || *_connectMethod == UnsecureFirst)
+				*_port = DAFILESRV_PORT;
+			else
+				*_port = SECURE_DAFILESRV_PORT;
+		}
     }
-    return retrieved;
+
```
Comment (mckellyln) [inline system/jlib/jutil.cpp]: indentation seems more than typical

**Comment by mckellyln** [comment]
@RussWhitehead add critical sections to protect passphrases and comments around deprecated dafs certificate/privatekey/passphrase code and perhaps change indentation.

**Comment by RussWhitehead** [comment]
@mckellyln Please rereview.

**Comment by RussWhitehead** [comment]
@mckellyln Please rereview.

**Review by mckellyln — APPROVED** [review]
Approved

**Comment by mckellyln** [comment]
@RussWhitehead looks good for merge.

**Comment by RussWhitehead** [comment]
@richardkchapman Please merge

**Pull request merged**